plugin-management-library/src/main/java/io/jenkins/tools/pluginmanager/impl/PluginManager.java

```diff
@@ -334,6 +369,30 @@ private void sortEffectivePlugins(Map<String, Plugin> effectivePlugins,
         }
     }
 
+    /**
+     * Writes the plugins and their versions to a plugin-lock.txt file.
+     *
+     * @param pluginsToDownload List of plugins that will be downloaded.
+     */
+    public void writePluginLockFile(List<Plugin> pluginsToDownload) {
+        if (pluginLockFile != null) {
+            try (BufferedWriter writer = new BufferedWriter(new FileWriter(pluginLockFile,StandardCharsets.UTF_8))) {
+                for (Plugin plugin : pluginsToDownload) {
+                    String pluginLine = String.format("%s:%s", plugin.getName(), plugin.getVersion());
+                    writer.write(pluginLine);
+                    writer.newLine();
+                }
+                logVerbose("Plugin lock file (" + pluginLockFile + ") has been successfully created.");
+            } catch (UnsupportedEncodingException e) {
+                e.printStackTrace();
+            } catch (IOException e) {
+                e.printStackTrace();
```

Comment:
Same use `logger.info("Your Context message", e);` through `import org.apache.log4j.Logger;`  instead of `e.printStackTrace();`